redaxo/src/addons/install/lib/install.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * @package redaxo\install
+ */
+class rex_install
+{
+    /**
+     * Download an AddOn from redaxo.org.
+     *
+     * @param string $addonKey e.g. "yform"
+     * @param string $version  e.g. "3.2.1"
+     *
+     * @throws rex_functional_exception
+     *
+     * @return bool|string
+     */
+    public function downloadAddon(string $addonKey, string $version)
+    {
+        if (rex_addon::exists($addonKey)) {
+            throw new rex_functional_exception(sprintf('AddOn "%s" already exists!', $addonKey));
```

Comment:
@staabm Blickst du noch durch, wann wir `rex_functional_exception` nutzen, und wann nicht?
Eigentlich war es glaube ich mal mehr gedacht, für über rex_i18n übersetzte Fehlermeldung, die dem User als roter Fehlerblock, also als legitim auftretbarer Fehler, angezeigt werden sollten.

Keine Ahnung, ob hier `rex_functional_exception` nun sinnvoll ist, oder nicht.

Reply:
wenn ich mir das so anschaue, haben wir zum großteil i18n strings. ein paar, die keinen haben.
keine ahnung, was hier "richtig" wäre.

oder man nimmt diese strings mit in i18n auf.

Reply:
Ich tendiere eigentlich eher dazu, dass es hier keine `rex_functional_exception`s sein sollten, sondern `rex_exception` oder irgendeine von den nativen von PHP.

Aber wie gesagt, mir ist es auch nicht mehr wirklich klar, wann wir was verwenden.

Reply:
Functional Exceptio ist für user-friendly errors, ja.

Reply:
habs jetzt auf `rex_exception` umgestellt.

Reply:
> Aber wie gesagt, mir ist es auch nicht mehr wirklich klar, wann wir was verwenden.

Könnten wir nicht eine kleine Übersicht hinterlegen, wann man welche Exception nutzen sollte?